// @chakra-ui/accordion/dist/esm/use-accordion.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { useDescendant, useDescendants } from "@chakra-ui/descendant";
import { useControllableState, useFocusEffect, useIds } from "@chakra-ui/hooks";
import { addItem, callAllHandlers, createOnKeyDown, getNextIndex, getPrevIndex, getValidChildren, isArray, mergeRefs, removeItem, createContext } from "@chakra-ui/utils";
import { useCallback, cloneElement, useRef, useState } from "react";
import * as warn from "./warning";

/**
 * useAccordion hook provides all the state and focus management logic
 * for accordion items.
 */
export function useAccordion(props) {
  var {
    onChange,
    defaultIndex,
    index: indexProp,
    allowMultiple,
    allowToggle,
    children
  } = props,
      htmlProps = _objectWithoutPropertiesLoose(props, ["onChange", "defaultIndex", "index", "allowMultiple", "allowToggle", "children"]); // validate the props and `warn` if used incorrectly


  warn.allowMultiple(props);
  warn.allowMultipleAndAllowToggle(props);
  /**
   * Think of this as the register to each accordion item.
   * We used to manage focus between accordion item buttons.
   *
   * Every accordion item, registers their button refs in this context
   */

  var domContext = useDescendants();
  /**
   * This state is used to track the index focused accordion
   * button when click on the button, tab on the button, or
   * use the down/up arrow to navigate.
   */

  var [focusedIndex, setFocusedIndex] = useState(-1);
  /**
   * Hook that manages the controlled and un-controlled state
   * for the accordion.
   */

  var [index, setIndex] = useControllableState({
    value: indexProp,
    defaultValue: () => {
      if (allowMultiple) {
        return defaultIndex != null ? defaultIndex : [];
      } else {
        return defaultIndex != null ? defaultIndex : -1;
      }
    },
    onChange,
    propsMap: {
      value: "index",
      defaultValue: "defaultIndex"
    }
  });
  /**
   * Filter out invalid children (null, false), in the case
   * of conditional rendering
   */

  var validChildren = getValidChildren(children);
  /**
   * Clone the accordion items and pass them the `onChange`
   * and `isOpen`
   */

  var _children = validChildren.map((child, idx) => {
    var isExpanded = isArray(index) ? index.includes(idx) : index === idx;
    return /*#__PURE__*/cloneElement(child, {
      isOpen: isExpanded,
      onChange: isOpen => {
        if (allowMultiple && isArray(index)) {
          var nextState = isOpen ? addItem(index, idx) : removeItem(index, idx);
          setIndex(nextState);
        } else {
          if (isOpen) {
            setIndex(idx);
          } else if (allowToggle) {
            setIndex(-1);
          }
        }
      }
    });
  });

  return {
    children: _children,
    htmlProps,
    focusedIndex,
    setFocusedIndex,
    domContext
  };
}
var [AccordionProvider, useAccordionContext] = createContext({
  name: "AccordionContext",
  errorMessage: "useAccordionContext: `context` is undefined. Seems you forgot to wrap the accordion components in `<Accordion />`"
});
export { AccordionProvider, useAccordionContext };

/**
 * useAccordionItem
 *
 * React hook that provides the open/close functionality
 * for an accordion item and it's children
 */
export function useAccordionItem(props) {
  var {
    isDisabled,
    isFocusable,
    onChange,
    isOpen,
    id
  } = props,
      htmlProps = _objectWithoutPropertiesLoose(props, ["isDisabled", "isFocusable", "onChange", "isOpen", "id"]);

  var {
    domContext,
    focusedIndex,
    setFocusedIndex
  } = useAccordionContext();
  var {
    descendants
  } = domContext;

  var onOpen = () => {
    onChange == null ? void 0 : onChange(true);
  };

  var onClose = () => {
    onChange == null ? void 0 : onChange(false);
  };

  var buttonRef = useRef(null);
  /**
   * Generate unique ids for all accordion item components (button and panel)
   */

  var [buttonId, panelId] = useIds(id, "accordion-button", "accordion-panel");
  warn.focusableNotDisabled(props);
  /**
   * Think of this as a way to register this accordion item
   * with it's parent `useAccordion`
   */

  var index = useDescendant({
    element: buttonRef.current,
    context: domContext,
    disabled: isDisabled,
    focusable: isFocusable
  });
  var shouldFocus = index === focusedIndex;
  /**
   * Autofocus the accordion button when
   * the active index matched the accordion item's index
   */

  useFocusEffect(buttonRef, {
    shouldFocus
  });
  /**
   * Toggle the visibility of the accordion item
   */

  var onClick = useCallback(() => {
    onChange == null ? void 0 : onChange(!isOpen);
    setFocusedIndex(index);
  }, [index, isOpen, onChange, setFocusedIndex]);
  /**
   * Manage keyboard navigation between accordion items.
   * `createOnKeyDown` makes it easy to write actions
   * for each event key
   */

  var onKeyDown = createOnKeyDown({
    keyMap: {
      ArrowDown: () => {
        var _nextAccordion$elemen;

        var nextIndex = getNextIndex(index, descendants.length);
        var nextAccordion = descendants[nextIndex];
        nextAccordion == null ? void 0 : (_nextAccordion$elemen = nextAccordion.element) == null ? void 0 : _nextAccordion$elemen.focus();
      },
      ArrowUp: () => {
        var _prevAccordion$elemen;

        var prevIndex = getPrevIndex(index, descendants.length);
        var prevAccordion = descendants[prevIndex];
        prevAccordion == null ? void 0 : (_prevAccordion$elemen = prevAccordion.element) == null ? void 0 : _prevAccordion$elemen.focus();
      },
      Home: () => {
        var _firstAccordion$eleme;

        var firstAccordion = descendants[0];
        firstAccordion == null ? void 0 : (_firstAccordion$eleme = firstAccordion.element) == null ? void 0 : _firstAccordion$eleme.focus();
      },
      End: () => {
        var _lastAccordion$elemen;

        var lastAccordion = descendants[descendants.length - 1];
        lastAccordion == null ? void 0 : (_lastAccordion$elemen = lastAccordion.element) == null ? void 0 : _lastAccordion$elemen.focus();
      }
    }
  });
  /**
   * Since each accordion item's button still remains tabbable, let's
   * update the focusedIndex when it receives focus
   */

  var onFocus = useCallback(() => setFocusedIndex(index), [index, setFocusedIndex]);
  var getButtonProps = useCallback(function (props, ref) {
    if (props === void 0) {
      props = {};
    }

    if (ref === void 0) {
      ref = null;
    }

    return _extends({}, props, {
      ref: mergeRefs(buttonRef, ref),
      id: buttonId,
      disabled: !!isDisabled,
      "aria-expanded": !!isOpen,
      "aria-controls": panelId,
      onClick: callAllHandlers(props.onClick, onClick),
      onFocus: callAllHandlers(props.onFocus, onFocus),
      onKeyDown: callAllHandlers(props.onKeyDown, onKeyDown)
    });
  }, [buttonId, isDisabled, isOpen, onClick, onFocus, onKeyDown, panelId]);
  var getPanelProps = useCallback(function (props, ref) {
    if (props === void 0) {
      props = {};
    }

    if (ref === void 0) {
      ref = null;
    }

    return _extends({}, props, {
      ref,
      role: "region",
      id: panelId,
      "aria-labelledby": buttonId,
      hidden: !isOpen
    });
  }, [buttonId, isOpen, panelId]);
  return {
    isOpen,
    isDisabled,
    isFocusable,
    onOpen,
    onClose,
    getButtonProps,
    getPanelProps,
    htmlProps
  };
}
