// @chakra-ui/theme/dist/esm/components/number-input.js
import { mode, multiStyleConfig } from "@chakra-ui/theme-tools";
import input from "./input";
var numberInput = multiStyleConfig({
  parts: {
    field: "the input field",
    stepper: "desktop - the increment and decrement button",
    stepperGroup: "desktop - the increment and decrement button group"
  },
  baseStyle: function baseStyle(props) {
    var _input$baseStyle;

    return {
      field: (_input$baseStyle = input.baseStyle) == null ? void 0 : _input$baseStyle.field,
      stepperGroup: {
        width: "24px"
      },
      stepper: {
        borderLeft: "1px solid",
        borderColor: mode("inherit", "whiteAlpha.300")(props),
        color: mode("inherit", "whiteAlpha.800")(props),
        _active: {
          bg: mode("gray.200", "whiteAlpha.300")(props)
        },
        _disabled: {
          opacity: 0.4,
          cursor: "not-allowed"
        }
      }
    };
  },
  sizes: {
    sm: getSize("sm"),
    md: getSize("md"),
    lg: getSize("lg")
  },
  variants: input.variants,
  defaultProps: input.defaultProps
});

function getSize(size) {
  var _input$sizes;

  var sizeStyle = (_input$sizes = input.sizes) == null ? void 0 : _input$sizes[size];
  var radius = {
    lg: "md",
    md: "md",
    sm: "sm"
  };
  return {
    field: sizeStyle == null ? void 0 : sizeStyle.field,
    stepper: {
      fontSize: size === "lg" ? "14px" : "10px",
      _first: {
        borderTopRightRadius: radius[size]
      },
      _last: {
        borderBottomRightRadius: radius[size],
        mt: "-1px",
        borderTopWidth: 1
      }
    }
  };
}

export default numberInput;
