// @chakra-ui/theme/dist/esm/components/select.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { multiStyleConfig, mode } from "@chakra-ui/theme-tools";
import input from "./input";
var select = multiStyleConfig({
  parts: {
    field: "the select field itself",
    icon: "the select field icon"
  },
  baseStyle: props => {
    var _input$baseStyle;

    return {
      field: _extends({}, (_input$baseStyle = input.baseStyle) == null ? void 0 : _input$baseStyle.field, {
        appearance: "none",
        paddingBottom: "1px",
        lineHeight: "normal",
        "> option": {
          bg: mode("white", "gray.700")(props)
        }
      }),
      icon: {
        color: "currentColor",
        fontSize: "1.25rem",
        _disabled: {
          opacity: 0.5
        }
      }
    };
  },
  sizes: input.sizes,
  variants: input.variants,
  defaultProps: input.defaultProps
});
export default select;
