// @chakra-ui/core/dist/cjs/chakra-provider.js
"use strict";

exports.__esModule = true;
exports.ChakraProvider = void 0;

var _cssReset = _interopRequireDefault(require("@chakra-ui/css-reset"));

var _portal = require("@chakra-ui/portal");

var _system = require("@chakra-ui/system");

var React = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * The global provider that must be added to make all Chakra components
 * work correctly
 */
var ChakraProvider = function ChakraProvider(props) {
  var _theme$config, _theme$config2;

  var theme = props.theme,
      children = props.children,
      storageManager = props.storageManager,
      resetCSS = props.resetCSS,
      portalConfig = props.portalConfig;

  if (!theme) {
    throw Error("ChakraProvider: the `theme` prop is required");
  }

  return /*#__PURE__*/React.createElement(_system.ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(_system.ColorModeProvider, {
    defaultValue: theme == null ? void 0 : (_theme$config = theme.config) == null ? void 0 : _theme$config.initialColorMode,
    useSystemColorMode: theme == null ? void 0 : (_theme$config2 = theme.config) == null ? void 0 : _theme$config2.useInitialColorMode,
    storageManager: storageManager
  }, /*#__PURE__*/React.createElement(_system.GlobalStyle, null), resetCSS && /*#__PURE__*/React.createElement(_cssReset["default"], null), portalConfig ? /*#__PURE__*/React.createElement(_portal.PortalManager, {
    zIndex: portalConfig == null ? void 0 : portalConfig.zIndex
  }, children) : children));
};

exports.ChakraProvider = ChakraProvider;
