// @chakra-ui/theme/dist/cjs/components/alert.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var alert = (0, _themeTools.multiStyleConfig)({
  parts: {
    container: "the alert container",
    title: "the alert title",
    icon: "the alert icon"
  },
  baseStyle: {
    container: {
      px: 4,
      py: 3
    },
    title: {
      fontWeight: "bold",
      lineHeight: "normal"
    },
    icon: {
      mr: 3,
      w: 5,
      h: 5
    }
  },
  variants: {
    subtle: function subtle(props) {
      var c = props.colorScheme;
      return {
        container: {
          bg: getBg(props)
        },
        icon: {
          color: (0, _themeTools.mode)(c + ".500", c + ".200")(props)
        }
      };
    },
    "left-accent": function leftAccent(props) {
      var c = props.colorScheme;
      return {
        container: {
          pl: 3,
          borderLeft: "4px solid",
          borderColor: (0, _themeTools.mode)(c + ".500", c + ".200")(props),
          bg: getBg(props)
        },
        icon: {
          color: (0, _themeTools.mode)(c + ".500", c + ".200")(props)
        }
      };
    },
    "top-accent": function topAccent(props) {
      var c = props.colorScheme;
      return {
        container: {
          pt: 2,
          borderTop: "4px solid",
          borderColor: (0, _themeTools.mode)(c + ".500", c + ".200")(props),
          bg: getBg(props)
        },
        icon: {
          color: (0, _themeTools.mode)(c + ".500", c + ".200")(props)
        }
      };
    },
    solid: function solid(props) {
      var c = props.colorScheme;
      return {
        container: {
          bg: (0, _themeTools.mode)(c + ".500", c + ".200")(props),
          color: (0, _themeTools.mode)("white", "gray.900")(props)
        }
      };
    }
  },
  defaultProps: {
    variant: "subtle"
  }
});

function getBg(props) {
  var theme = props.theme,
      c = props.colorScheme;
  var lightBg = (0, _themeTools.getColor)(theme, c + ".100", c);
  var darkBg = (0, _themeTools.transparentize)(c + ".200", 0.16)(theme);
  return (0, _themeTools.mode)(lightBg, darkBg)(props);
}

var _default = alert;
exports["default"] = _default;
