// @chakra-ui/theme/dist/cjs/components/close-button.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var closeButton = (0, _themeTools.styleConfig)({
  baseStyle: function baseStyle(props) {
    var hoverBg = (0, _themeTools.mode)("blackAlpha.100", "whiteAlpha.100")(props);
    var activeBg = (0, _themeTools.mode)("blackAlpha.200", "whiteAlpha.200")(props);
    return {
      borderRadius: "md",
      transition: "all 0.2s",
      _disabled: {
        opacity: 0.4,
        cursor: "not-allowed",
        boxShadow: "none"
      },
      _hover: {
        bg: hoverBg
      },
      _active: {
        bg: activeBg
      },
      _focus: {
        boxShadow: "outline"
      }
    };
  },
  sizes: {
    lg: {
      w: "40px",
      h: "40px",
      fontSize: "16px"
    },
    md: {
      w: "32px",
      h: "32px",
      fontSize: "12px"
    },
    sm: {
      w: "24px",
      h: "24px",
      fontSize: "10px"
    }
  },
  defaultProps: {
    size: "md"
  }
});
var _default = closeButton;
exports["default"] = _default;
