// @chakra-ui/theme/dist/cjs/components/menu.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var menu = (0, _themeTools.multiStyleConfig)({
  parts: {
    item: "the menu item button",
    command: "the menu item command",
    list: "the menu items wrapper",
    button: "the menu list trigger",
    groupTitle: "the menu group title"
  },
  baseStyle: function baseStyle(props) {
    return {
      list: {
        bg: (0, _themeTools.mode)("#fff", "gray.700")(props),
        boxShadow: (0, _themeTools.mode)("sm", "dark-lg")(props),
        color: "inherit",
        minW: "3xs",
        py: "2",
        zIndex: 1,
        borderRadius: "md",
        borderWidth: "1px"
      },
      item: {
        py: "0.4rem",
        px: "0.8rem",
        transition: "background 50ms ease-in 0s",
        _focus: {
          bg: (0, _themeTools.mode)("gray.100", "whiteAlpha.100")(props)
        },
        _active: {
          bg: (0, _themeTools.mode)("gray.200", "whiteAlpha.200")(props)
        },
        _expanded: {
          bg: (0, _themeTools.mode)("gray.100", "whiteAlpha.100")(props)
        },
        _disabled: {
          opacity: 0.4,
          cursor: "not-allowed"
        }
      },
      groupTitle: {
        mx: 4,
        my: 2,
        fontWeight: "semibold",
        fontSize: "sm"
      },
      command: {
        opacity: 0.6
      }
    };
  }
});
var _default = menu;
exports["default"] = _default;
