// @chakra-ui/theme/dist/cjs/components/number-input.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var _input = _interopRequireDefault(require("./input"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var numberInput = (0, _themeTools.multiStyleConfig)({
  parts: {
    field: "the input field",
    stepper: "desktop - the increment and decrement button",
    stepperGroup: "desktop - the increment and decrement button group"
  },
  baseStyle: function baseStyle(props) {
    var _input$baseStyle;

    return {
      field: (_input$baseStyle = _input["default"].baseStyle) == null ? void 0 : _input$baseStyle.field,
      stepperGroup: {
        width: "24px"
      },
      stepper: {
        borderLeft: "1px solid",
        borderColor: (0, _themeTools.mode)("inherit", "whiteAlpha.300")(props),
        color: (0, _themeTools.mode)("inherit", "whiteAlpha.800")(props),
        _active: {
          bg: (0, _themeTools.mode)("gray.200", "whiteAlpha.300")(props)
        },
        _disabled: {
          opacity: 0.4,
          cursor: "not-allowed"
        }
      }
    };
  },
  sizes: {
    sm: getSize("sm"),
    md: getSize("md"),
    lg: getSize("lg")
  },
  variants: _input["default"].variants,
  defaultProps: _input["default"].defaultProps
});

function getSize(size) {
  var _input$sizes;

  var sizeStyle = (_input$sizes = _input["default"].sizes) == null ? void 0 : _input$sizes[size];
  var radius = {
    lg: "md",
    md: "md",
    sm: "sm"
  };
  return {
    field: sizeStyle == null ? void 0 : sizeStyle.field,
    stepper: {
      fontSize: size === "lg" ? "14px" : "10px",
      _first: {
        borderTopRightRadius: radius[size]
      },
      _last: {
        borderBottomRightRadius: radius[size],
        mt: "-1px",
        borderTopWidth: 1
      }
    }
  };
}

var _default = numberInput;
exports["default"] = _default;
