// @chakra-ui/theme/dist/cjs/components/select.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var _input = _interopRequireDefault(require("./input"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var select = (0, _themeTools.multiStyleConfig)({
  parts: {
    field: "the select field itself",
    icon: "the select field icon"
  },
  baseStyle: function baseStyle(props) {
    var _input$baseStyle;

    return {
      field: _extends({}, (_input$baseStyle = _input["default"].baseStyle) == null ? void 0 : _input$baseStyle.field, {
        appearance: "none",
        paddingBottom: "1px",
        lineHeight: "normal",
        "> option": {
          bg: (0, _themeTools.mode)("white", "gray.700")(props)
        }
      }),
      icon: {
        color: "currentColor",
        fontSize: "1.25rem",
        _disabled: {
          opacity: 0.5
        }
      }
    };
  },
  sizes: _input["default"].sizes,
  variants: _input["default"].variants,
  defaultProps: _input["default"].defaultProps
});
var _default = select;
exports["default"] = _default;
