// @chakra-ui/theme/dist/cjs/components/textarea.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _themeTools = require("@chakra-ui/theme-tools");

var _input = _interopRequireDefault(require("./input"));

var _input$baseStyle, _input$sizes, _input$sizes2, _input$sizes3, _input$variants4;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var textarea = (0, _themeTools.styleConfig)({
  baseStyle: _extends({}, (_input$baseStyle = _input["default"].baseStyle) == null ? void 0 : _input$baseStyle.field, {
    paddingY: "8px",
    minHeight: "80px",
    lineHeight: "short"
  }),
  sizes: {
    sm: (_input$sizes = _input["default"].sizes) == null ? void 0 : _input$sizes.sm.field,
    md: (_input$sizes2 = _input["default"].sizes) == null ? void 0 : _input$sizes2.md.field,
    lg: (_input$sizes3 = _input["default"].sizes) == null ? void 0 : _input$sizes3.lg.field
  },
  variants: {
    outline: function outline(props) {
      var _input$variants$outli, _input$variants, _input$variants$outli2;

      return (_input$variants$outli = (_input$variants = _input["default"].variants) == null ? void 0 : (_input$variants$outli2 = _input$variants.outline(props)) == null ? void 0 : _input$variants$outli2.field) != null ? _input$variants$outli : {};
    },
    flushed: function flushed(props) {
      var _input$variants$flush, _input$variants2, _input$variants2$flus;

      return (_input$variants$flush = (_input$variants2 = _input["default"].variants) == null ? void 0 : (_input$variants2$flus = _input$variants2.flushed(props)) == null ? void 0 : _input$variants2$flus.field) != null ? _input$variants$flush : {};
    },
    filled: function filled(props) {
      var _input$variants$fille, _input$variants3;

      return (_input$variants$fille = (_input$variants3 = _input["default"].variants) == null ? void 0 : _input$variants3.filled(props).field) != null ? _input$variants$fille : {};
    },
    unstyled: (_input$variants4 = _input["default"].variants) == null ? void 0 : _input$variants4.unstyled.field
  },
  defaultProps: {
    size: "md",
    variant: "outline"
  }
});
var _default = textarea;
exports["default"] = _default;
